File: clients/node/sdk/models/familiarFollowers.ts
import {createAccountFromDiscriminatorValue} from './createAccountFromDiscriminatorValue';
import {Account} from './index';
import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class FamiliarFollowers implements AdditionalDataHolder, Parsable {
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
    private _additionalData: Record<string, unknown>;
    /** The followers property */
    private _followers?: Account[] | undefined;
    /** The id property */
    private _id?: string | undefined;
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    public get additionalData() {
        return this._additionalData;
    };
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public set additionalData(value: Record<string, unknown>) {
        this._additionalData = value;
    };
    /**
     * Instantiates a new FamiliarFollowers and sets the default values.
     */
    public constructor() {
        this._additionalData = {};
    };
    /**
     * Gets the followers property value. The followers property
     * @returns a Account
     */
    public get followers() {
        return this._followers;
    };
    /**
     * Sets the followers property value. The followers property
     * @param value Value to set for the followers property.
     */
    public set followers(value: Account[] | undefined) {
        this._followers = value;
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {
            "followers": n => { this.followers = n.getCollectionOfObjectValues<Account>(createAccountFromDiscriminatorValue); },
            "id": n => { this.id = n.getStringValue(); },
        };
    };
    /**
     * Gets the id property value. The id property
     * @returns a string
     */
    public get id() {
        return this._id;
    };
    /**
     * Sets the id property value. The id property
     * @param value Value to set for the id property.
     */
    public set id(value: string | undefined) {
        this._id = value;
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        writer.writeCollectionOfObjectValues<Account>("followers", this.followers);
        writer.writeStringValue("id", this.id);
        writer.writeAdditionalData(this.additionalData);
    };
}

File: clients/python/sdk/models/familiar_followers.py
from __future__ import annotations
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from kiota_abstractions.utils import lazy_import
from typing import Any, Callable, Dict, List, Optional, Union

account = lazy_import('mastodon_client_lib.models.account')

class FamiliarFollowers(AdditionalDataHolder, Parsable):
    @property
    def additional_data(self,) -> Dict[str, Any]:
        """
        Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Returns: Dict[str, Any]
        """
        return self._additional_data
    
    @additional_data.setter
    def additional_data(self,value: Dict[str, Any]) -> None:
        """
        Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Args:
            value: Value to set for the AdditionalData property.
        """
        self._additional_data = value
    
    def __init__(self,) -> None:
        """
        Instantiates a new FamiliarFollowers and sets the default values.
        """
        # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        self._additional_data: Dict[str, Any] = {}

        # The followers property
        self._followers: Optional[List[account.Account]] = None
        # The id property
        self._id: Optional[str] = None
    
    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> FamiliarFollowers:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: FamiliarFollowers
        """
        if parse_node is None:
            raise Exception("parse_node cannot be undefined")
        return FamiliarFollowers()
    
    @property
    def followers(self,) -> Optional[List[account.Account]]:
        """
        Gets the followers property value. The followers property
        Returns: Optional[List[account.Account]]
        """
        return self._followers
    
    @followers.setter
    def followers(self,value: Optional[List[account.Account]] = None) -> None:
        """
        Sets the followers property value. The followers property
        Args:
            value: Value to set for the followers property.
        """
        self._followers = value
    
    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
            "followers": lambda n : setattr(self, 'followers', n.get_collection_of_object_values(account.Account)),
            "id": lambda n : setattr(self, 'id', n.get_str_value()),
        }
        return fields
    
    @property
    def id(self,) -> Optional[str]:
        """
        Gets the id property value. The id property
        Returns: Optional[str]
        """
        return self._id
    
    @id.setter
    def id(self,value: Optional[str] = None) -> None:
        """
        Sets the id property value. The id property
        Args:
            value: Value to set for the id property.
        """
        self._id = value
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if writer is None:
            raise Exception("writer cannot be undefined")
        writer.write_collection_of_object_values("followers", self.followers)
        writer.write_str_value("id", self.id)
        writer.write_additional_data_value(self.additional_data)
    

